cmd/kube-proxy/app/server_others.go

```diff
@@ -91,9 +91,40 @@ func newProxyServer(
 		return nil, fmt.Errorf("unable to register configz: %s", err)
 	}
 
+	hostname, err := utilnode.GetHostname(config.HostnameOverride)
+	if err != nil {
+		return nil, err
+	}
+
+	client, eventClient, err := createClients(config.ClientConnection, master)
+	if err != nil {
+		return nil, err
+	}
+
+	// kube-proxy needs to detect the IP family
+	// because iptables has a different binary per IP family
+
+	// The IP family will be determined by the nodeIP used by the proxier
+	// The order of precedence is:
+	// 1. config.bindAddress if bindAddress is not 0.0.0.0 or ::
+	// 2. based on the priority: NodeInternalIP, NodeExternalIP
+	//    the IP of the Node with the same nodename:
+	//  2.1 the provided hostname
+	//  2.2 the OS hostname
+	// 3. if no IP is found it defaults to 127.0.0.1 and IPv4
+
+	nodeIP := net.ParseIP(config.BindAddress)
+	if nodeIP.IsUnspecified() {
+		nodeIP = utilnode.GetNodeIP(client, hostname)
+		if nodeIP == nil {
+			klog.V(0).Infof("can't determine this node's IP, assuming 127.0.0.1; if this is incorrect, please set the --bind-address flag")
+			nodeIP = net.ParseIP("127.0.0.1")
+		}
+	}
+
 	protocol := utiliptables.ProtocolIPv4
-	if net.ParseIP(config.BindAddress).To4() == nil {
-		klog.V(0).Infof("IPv6 bind address (%s), assume IPv6 operation", config.BindAddress)
+	if nodeIP.To4() == nil {
```

Comment:
can you update that to use `utilnet.IsIPv6()`?

Reply:
done